src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -47,137 +48,138 @@ public static bool IsInsideOrchestrator(SyntaxNode node)
             return false;
         }
 
-        internal static bool TryGetClassSymbol(SyntaxNode node, SemanticModel semanticModel, out INamedTypeSymbol classSymbol)
+        public static bool IsMarkedDeterministic(SyntaxNode node) => TryGetDeterministicAttribute(node, out _);
+
+
+        public static bool TryGetDeterministicAttribute(SyntaxNode node, out SyntaxNode deterministicAttribute) => TryGetAttribute(node, "Deterministic", out deterministicAttribute);
+
+        private static bool TryGetAttribute(SyntaxNode node, string attributeName, out SyntaxNode attribute)
         {
-            var currNode = node.IsKind(SyntaxKind.ClassDeclaration) ? node : node.Parent;
-            while (!currNode.IsKind(SyntaxKind.ClassDeclaration))
+            if (TryGetMethodDeclaration(node, out SyntaxNode methodDeclaration))
             {
-                if (currNode.IsKind(SyntaxKind.CompilationUnit))
+                var attributeLists = methodDeclaration.ChildNodes().Where(x => x.IsKind(SyntaxKind.AttributeList));
+                foreach (var attributeList in attributeLists)
                 {
-                    classSymbol = null;
-                    return false;
+                    attribute = attributeList.ChildNodes().First();
```

Comment:
Is `ChildNodes()` always guaranteed to have at least one element?

Reply:
Yes, I will add that as a comment